packages/w3up-client/README.md

```diff
@@ -227,25 +250,122 @@ As mentioned, UCAN opens up a number of options in how to integrate with w3up: S
 
 You can implement each of these in a number of ways, but we talk through some considerations when implementing a given option.
 
-1. Client-server
+#### Client-server
+```mermaid
+sequenceDiagram
+    participant User
+    w3up-client in backend->>w3up-client in backend: Client set with Agent with delegation from Space
+    User->>w3up-client in backend: Upload data
+    w3up-client in backend->>web3.storage w3up service: Upload data
+```
 - For your backend to be scalable, you might consider using serverless workers or a queue in front of a server
-- In either case, it's likely easiest to create and register your Space using [w3cli](https://github.com/web3-storage/w3cli) rather than using `w3up-client` to do so
-- You can then generate your own Agent and delegate the ability to upload to your Space using something like [this example](#bringing-your-own-agent-and-delegation)
-- Once your user uploads data to your backend, you can run any of the `upload` methods
-
-2. Delegated
+- In either case, you'll need a registered Space, and your client instance in your backend to have an Agent with a delegation from this Space
+  - (Recommended) It's likely easiest to create and register your Space using [w3cli](https://github.com/web3-storage/w3cli) rather than using `w3up-client` to do so (especially if your backend isn't persistent); you can then generate your own Agent and delegate the ability to upload to your Space using something like [this example](#bringing-your-own-agent-and-delegation)
+  - If your backend is persistent, you can do this or do everything in the client directly ([create Space](#creating-and-registering-spaces) and [get delegation](#delegating-from-space-to-agent))
+- After this, once your user uploads data to your backend, you can run any of the `upload` methods
+
+#### Delegated
+```mermaid
+sequenceDiagram
+    participant w3up-client in user
+    participant w3up-client in backend
+    participant web3.storage w3up service
+    w3up-client in backend->>w3up-client in backend: Client set with Agent and delegation from Space
```

Comment:
```suggestion
    w3up-client in backend->>w3up-client in backend: Client created with Agent and delegation from Space
```

Reply:
might not be understanding what you're trying to say here, but "set" feels confusing..